### PR TITLE
[FLINK-18842][e2e] Added 10min timeout to building the container.

### PR DESCRIPTION
**Pull request by XComp**
## What is the purpose of the change

The docker build of `test_docker_embedded_job.sh` is failing once in a while. We haven't come up with a reason for it, yet. We also struggled to reproduce it. The discussion is happening in FLINK-18842.

## Brief change log

For now, I added a timeout to the docker build to enable the retry that is already in place.

## Verifying this change

This change is already covered by existing tests, such as `test_docker_embedded_job.sh`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 1]
```diff
@@ -830,3 +830,30 @@ run_test_with_timeout() {
...
+
+  # invoke command
+  (eval "$command"; rm $pidFile; ) &
+  pid=$!
+  echo $pid > $pidFile
+
+  # invoke timeout guard
+  (
+    sleep $timeout
+    if [[ -e $pidFile ]]; then
```
Comment (zentol) [inline flink-end-to-end-tests/test-scripts/common.sh:849]: We should be able to re-use `run_test_with_timeout`. All we basically need to do is move this
``` 
echo "Printing Flink logs and killing it:"
cat ${FLINK_DIR}/log/*
```
into a separate function, and add a new `run_with_timeout` that accepts a timeout, run-command, and on-failure-command.

```
run_test_with_timeout() {
  local TEST_TIMEOUT_SECONDS=$1
  shift
  local TEST_COMMAND=$@
  internal_run_with_timeout $TEST_TIMEOUT_SECONDS $TEST_COMMAND print_logs
}

run_with_timeout() {
  internal_run_with_timeout $1 $2 ""
}

internal_run_with_timeout() {
  local timeout=$1
  local runCommand=$2
  local onFailureCommand=$3

  // do stuff
}

print_logs() {
  echo "Printing Flink logs and killing it:"
   cat ${FLINK_DIR}/log/*
}
```

Comment (XComp) [reply flink-end-to-end-tests/test-scripts/common.sh]: First, I tried to make it work with the old implementation. But I struggled to test it locally on MacOS. Hence, I switched to the new implementation. I tried to make the code change not cause changes to test output. Let's see if the tests run through.

**Comment by zentol** [comment]
`/home/vsts/work/1/s/flink-end-to-end-tests/test-scripts/common_utils.sh: line 33: kill_testing_watchdog: command not found`

File: flink-end-to-end-tests/test-scripts/common_docker.sh [hunk 2]
```diff
@@ -50,7 +50,7 @@ function build_image() {
     ./add-custom.sh -u localhost:9999/flink.tgz -n ${image_name}
 
     echo "Building images"
-    docker build --no-cache --network="host" -t ${image_name} dev/${image_name}-debian
+    run_with_timeout 600 "${TEST_DATA_DIR}/docker-build-${RANDOM}.pid" docker build --no-cache --network="host" -t ${image_name} dev/${image_name}-debian
```
Comment (zentol) [inline flink-end-to-end-tests/test-scripts/common_docker.sh:53]:
```suggestion
    run_with_timeout 600 docker build --no-cache --network="host" -t ${image_name} dev/${image_name}-debian
```

File: flink-end-to-end-tests/test-scripts/common.sh [hunk 3]
```diff
@@ -801,32 +801,44 @@ function extract_job_id_from_job_submission_return() {
...
 
   (
-      cmdpid=$BASHPID
-      (sleep $TEST_TIMEOUT_SECONDS # set a timeout for this test
-      echo "Test (pid: $cmdpid) did not finish after $TEST_TIMEOUT_SECONDS seconds."
-      echo "Printing Flink logs and killing it:"
-      cat ${FLINK_DIR}/log/*
-      kill "$cmdpid") & watchdog_pid=$!
+      command_pid=$BASHPID
+      (sleep "${timeout_in_seconds}" # set a timeout for this test
```
Comment (zentol) [inline flink-end-to-end-tests/test-scripts/common.sh:823]:
```suggestion
      (sleep "${timeout_in_seconds}" # set a timeout for this command
```

**Pull request merged**